connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/ProcessInstanceContext.java

```diff
@@ -18,9 +18,43 @@
 
 import java.util.Map;
 
+/**
+ * Represents the context for a specific process instance. This interface provides a contract to
+ * interact with, modify, and retrieve information related to a particular instance of a process.
+ */
 public interface ProcessInstanceContext {
 
+  /**
+   * Retrieves the unique key associated with the process instance.
+   *
+   * @return A unique identifier representing the process instance.
+   */
   Long getProcessInstanceKey();
 
-  Map<String, Object> getVariables();
+  /**
+   * Updates the context with new or modified variables. This method provides a way to make changes
+   * to the state of the process instance.
+   *
+   * @param variables A map containing variable names as keys and their corresponding values.
+   *     Existing variables with the same name will be overwritten.
+   */
+  void update(Map<String, String> variables);
```

Comment:
Why do you need to update any variables? The prociess instance context encapsules all the variable handling itself.